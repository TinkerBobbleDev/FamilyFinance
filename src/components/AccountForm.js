import React from "react";
import { withStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import { connect } from "react-redux";
import Paper from "@material-ui/core/Paper";

export const styles = theme => ({
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    marginTop: theme.spacing.unit,
    marginBottom: theme.spacing.unit,
    width: 250,
    padding: 10
  },
  button: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    width: 250,
    marginTop: theme.spacing.unit,
    marginBottom: theme.spacing.unit
  },
  paper: theme.mixins.gutters({
    paddingLeft: 0,
    paddingRight: 0,
    marginTop: theme.spacing.unit * 3,
    overflowX: "auto"
  })
});

export class AccountForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: props.account ? props.account.name : "",
      balance: props.account ? (props.account.balance / 100).toString() : "",
      error: "",
      submit_button_title: props.account ? "Update" : "Add",
      disableName: props.account ? true : false
    };
  }

  onSubmit = e => {
    e.preventDefault();

    if (!this.state.name || !this.state.balance) {
      this.setState(() => ({ error: "Please provide name and balance." }));
    } else {
      if (
        typeof this.props.accounts !== "undefined" &&
        this.props.accounts.length > 0
      ) {
        if (
          this.props.accounts.find(
            account => account.name === this.state.name
          ) === undefined
        ) {
          this.setState(() => ({ error: "" }));
          this.props.onSubmit({
            name: this.state.name,
            balance: parseFloat(this.state.balance, 10) * 100
          });
        } else {
          if (this.state.submit_button_title === "Update") {
            this.props.onSubmit({
              name: this.state.name,
              balance: parseFloat(this.state.balance, 10) * 100
            });
          } else {
            this.setState(() => ({
              name: "",
              balance: "",
              error: "This Account already exists"
            }));
          }
        }
      } else {
        this.setState(() => ({ error: "" }));
        this.props.onSubmit({
          name: this.state.name,
          balance: parseFloat(this.state.balance, 10) * 100
        });
      }
    }
  };

  onNameChange = e => {
    const name = e.target.value;
    this.setState(() => ({ name }));
  };

  onBalanceChange = e => {
    const balance = e.target.value;

    if (!balance || balance.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ balance }));
    }
  };

  render() {
    const { classes } = this.props;
    return (
      <Paper className={classes.paper} elevation={4}>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <TextField
            required
            disabled={this.state.disableName}
            className={classes.textField}
            type="text"
            placeholder="name"
            value={this.state.name}
            onChange={this.onNameChange}
          />
          <TextField
            required
            className={classes.textField}
            type="text"
            placeholder="Balance"
            value={this.state.balance}
            onChange={this.onBalanceChange}
          />

          <Button
            onClick={this.onSubmit}
            className={classes.button}
            variant="raised"
            color="primary"
          >
            {this.state.submit_button_title}
          </Button>
        </form>
      </Paper>
    );
  }
}
const mapStateToProps = (state, props) => {
  return { accounts: state.accounts };
};

export default connect(mapStateToProps)(withStyles(styles)(AccountForm));
