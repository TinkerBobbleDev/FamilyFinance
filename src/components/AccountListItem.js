import React from "react";
import numeral from "numeral";
import TableRow from "@material-ui/core/TableRow";
import TableCell from "@material-ui/core/TableCell";
import IconButton from "@material-ui/core/IconButton";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import MoreVertIcon from "@material-ui/icons/MoreVert";
import { withRouter } from "react-router-dom";

const ITEM_HEIGHT = 48;

export class AccountListItems extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      anchorEl: null
    };
  }

  handleClick = event => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  deleteAccount = () => {
    this.props.onDelete({ name: this.props.name });
  };

  editAccount = () => {
    this.props.history.push("/accounts/edit/" + this.props.name);
    this.setState({ anchorEl: null });
  };

  copyAccount = () => {
    this.props.onCopy(this.props.name);
    this.setState({ anchorEl: null });
  };

  render() {
    let open = Boolean(this.state.anchorEl);
    return (
      <TableRow key={this.props.name}>
        <TableCell>{this.props.name}</TableCell>
        <TableCell numeric>
          {numeral(this.props.balance / 100).format("â‚¬ 0,0.00")}
        </TableCell>
        <TableCell>
          <IconButton
            aria-label="More"
            aria-owns={open ? "long-menu" : null}
            aria-haspopup="true"
            onClick={this.handleClick}
          >
            <MoreVertIcon />
          </IconButton>
          <Menu
            id="long-menu"
            anchorEl={this.state.anchorEl}
            open={open}
            onClose={this.handleClose}
            PaperProps={{
              style: {
                maxHeight: ITEM_HEIGHT * 4.5,
                width: 200
              }
            }}
          >
            {/*<button onClick={this.deleteAccount}>Remove</button>
                        <Link to={`/accounts/edit/${this.props.id}`}>
                            <button >Edit</button>
                        </Link>*/}
            <MenuItem key="Edit" onClick={this.editAccount}>
              Edit
            </MenuItem>

            <MenuItem key="Remove" onClick={this.deleteAccount}>
              Remove
            </MenuItem>

            {/*<MenuItem key='Copy' onClick={this.copyAccount}>
                            Copy
                        </MenuItem>*/}
          </Menu>
        </TableCell>
      </TableRow>
    );
  }
}

export default withRouter(AccountListItems);
