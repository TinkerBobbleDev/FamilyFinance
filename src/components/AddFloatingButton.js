import React from "react";
import { withStyles } from "@material-ui/core/styles";
import AddIcon from "@material-ui/icons/Add";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import TransactionFrom from "./TransactionForm";
import { connect } from "react-redux";
import { startAddTransaction } from "../actions/transactions";
import { updateAccountBalance } from "../actions/accounts";
import withMobileDialog from "@material-ui/core/withMobileDialog";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";

export const styles = theme => ({
  floatingButton: {
    margin: 0,
    top: "auto",
    right: 20,
    bottom: 20,
    left: "auto",
    position: "fixed"
  },
  dialogContent: {
    textAlign: "center",
    maxWidth: 400
  },
  dialogTitle: {
    textAlign: "center"
  }
});

export class AddFloatingButton extends React.Component {
  state = {
    open: false
  };

  onSubmit = transaction => {
    if (transaction.type === "Transfer") {
      let firstTransaction = {
        type: transaction.type,
        description: transaction.description,
        amount: transaction.amount,
        category: transaction.category,
        date: transaction.date,
        account: transaction.transferFrom
      };
      let SecondTransaction = {
        type: transaction.type,
        description: transaction.description,
        amount: transaction.amount,
        category: transaction.category,
        date: transaction.date,
        account: transaction.transferTo
      };
      this.props.startAddTransfer(
        transaction,
        firstTransaction,
        SecondTransaction
      );
      return this.handleClose();
    }
    this.props.startAddTransaction(transaction);

    return this.handleClose();
  };

  handleClickFloatingButton = event => {
    this.setState({ open: true });
  };
  handleClose = () => {
    this.setState({ open: false });
  };
  render() {
    const { classes, fullScreen } = this.props;

    return (
      <div>
        <Button
          variant="fab"
          color="primary"
          aria-label="add"
          className={classes.floatingButton}
          onClick={this.handleClickFloatingButton}
        >
          <AddIcon />
        </Button>
        <Dialog
          fullScreen={fullScreen}
          open={this.state.open}
          onClose={this.handleClose}
          aria-labelledby="form-dialog-title"
        >
          <DialogTitle className={classes.dialogTitle} id="form-dialog-title">
            Add Transaction
          </DialogTitle>
          <DialogContent className={classes.dialogContent}>
            <TransactionFrom
              onSubmit={this.onSubmit}
              onClose={this.handleClose}
            />
          </DialogContent>
        </Dialog>
      </div>
    );
  }
}
const mapDispatchToProps = dispatch => ({
  startAddTransaction: transaction =>
    dispatch(startAddTransaction(transaction)).then(() => {
      let delta = transaction.amount;
      if (transaction.type === "Expense") {
        delta = -delta;
      }
      dispatch(updateAccountBalance(transaction.account, delta));
    }),
  startAddTransfer: (transaction, firstTransaction, secondTransaction) =>
    dispatch(startAddTransaction(transaction))
      .then(() => {
        dispatch(
          updateAccountBalance(firstTransaction.account, -transaction.amount)
        );
      })
      .then(() => {
        dispatch(
          updateAccountBalance(secondTransaction.account, transaction.amount)
        );
      })
});

export default connect(undefined, mapDispatchToProps)(
  withStyles(styles)(withMobileDialog()(AddFloatingButton))
);
