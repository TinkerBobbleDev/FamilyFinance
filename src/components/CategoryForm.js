import React from 'react';
import {withStyles} from 'material-ui/styles';
import TextField from 'material-ui/TextField';
import Button from 'material-ui/Button';
import {connect} from 'react-redux';
import Paper from 'material-ui/Paper';

const styles = theme => ({
    textField: {
        marginLeft: theme.spacing.unit,
        marginRight: theme.spacing.unit,
        marginTop: theme.spacing.unit,
        marginBottom: theme.spacing.unit,
        width: 250,
        padding: 10
    },
    button: {
        marginLeft: theme.spacing.unit,
        marginRight: theme.spacing.unit,
        width: 250,
        marginTop: theme.spacing.unit,
        marginBottom: theme.spacing.unit
    },
    paper: theme
        .mixins
        .gutters({

            paddingLeft: 0,
            paddingRight: 0,
            marginTop: theme.spacing.unit * 3,

            overflowX: 'auto'
        })
});

class CategoryForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            name: props.category
                ? props.category.name
                : '',
            error: '',
            submit_button_title: props.category
                ? 'Update'
                : 'Add'
        };
    };

    onSubmit = (e) => {
        e.preventDefault();
        if (!this.state.name) {
            this.setState(() => ({error: 'Please provide name'}));
        } else {
            if (typeof this.props.categories !== 'undefined' && this.props.categories.length > 0) {

                if (this.props.categories.find((category) => category.name === this.state.name) === undefined) {
                    this.setState(() => ({error: ''}));
                    this
                        .props
                        .onSubmit({name: this.state.name});
                } else {
                    this.setState(() => ({name: '', error: 'Category already exists'}));
                }
            } else {
                // First Added Category
                this.setState(() => ({error: ''}));
                this
                    .props
                    .onSubmit({
                        name: this.state.name,
                    });
            }
        }
    };

    onNameChange = (e) => {
        const name = e.target.value;
        this.setState(() => ({name}));
    };

    render() {
        const {classes} = this.props;
        return (
            <Paper className={classes.paper} elevation={4}>

                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>
                    <TextField
                        className={classes.textField}
                        type="text"
                        placeholder="category"
                        value={this.state.name}
                        onChange={this.onNameChange}/>

                    <Button
                        onClick={this.onSubmit}
                        className={classes.button}
                        variant="raised"
                        color="primary">{this.state.submit_button_title}</Button>
                </form>
            </Paper>
        )
    }

}

const mapStateToProps = (state, props) => {
    return {categories: state.categories};
};

export default connect(mapStateToProps)(withStyles(styles)(CategoryForm));