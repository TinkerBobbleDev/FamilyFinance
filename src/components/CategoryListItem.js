import React from "react";
import TableCell from "@material-ui/core/TableCell";
import TableRow from "@material-ui/core/TableRow";
import IconButton from "@material-ui/core/IconButton";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import MoreVertIcon from "@material-ui/icons/MoreVert";
import { withRouter } from "react-router-dom";

const ITEM_HEIGHT = 48;

export class CategoryListItem extends React.Component {
  state = {
    anchorEl: null
  };

  handleClick = event => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  deleteCategory = () => {
    this.props.onDelete({ id: this.props.id });
  };

  copyCategory = () => {
    this.props.onCopy(this.props.id);
    this.setState({ anchorEl: null });
  };

  render() {
    const open = Boolean(this.state.anchorEl);
    return (
      <TableRow key={this.props.id}>
        <TableCell>{this.props.name}</TableCell>
        <TableCell>
          <IconButton
            aria-label="More"
            aria-owns={open ? "long-menu" : null}
            aria-haspopup="true"
            onClick={this.handleClick}
          >
            <MoreVertIcon />
          </IconButton>
          <Menu
            id="long-menu"
            anchorEl={this.state.anchorEl}
            open={open}
            onClose={this.handleClose}
            PaperProps={{
              style: {
                maxHeight: ITEM_HEIGHT * 4.5,
                width: 200
              }
            }}
          >
            {/*<button onClick={this.deleteAccount}>Remove</button>
                        <Link to={`/accounts/edit/${this.props.id}`}>
                            <button >Edit</button>
                        </Link>*/}

            <MenuItem key="Remove" onClick={this.deleteCategory}>
              Remove
            </MenuItem>

            {/*<MenuItem key='Copy' onClick={this.copyCategory}>
                            Copy
                        </MenuItem>*/}
          </Menu>
        </TableCell>
      </TableRow>
    );
  }
}

export default withRouter(CategoryListItem);
