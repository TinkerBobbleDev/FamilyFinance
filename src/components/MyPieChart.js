import React from "react";
import { withStyles } from "@material-ui/core/styles";
import {
  PieChart,
  Pie,
  Cell,
  ResponsiveContainer,
  Legend,
  Tooltip
} from "recharts";
import Paper from "@material-ui/core/Paper";

export const styles = theme => ({
  root: {
    WebkitBoxSizing: "border-box",
    MozBoxSizing: "border-box",
    padding: 10,
    height: 380,
    backgroundColor: "#fff"
  },
  paper: theme.mixins.gutters({
    boxSizing: "border - box",
    paddingLeft: 16,
    paddingRight: 16,
    marginTop: theme.spacing.unit,
    overflowX: "auto"
  })
});

export class MyPieChart extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      colors: []
    };
  }

  getRandomColor = categoriesNumber => {
    var letters = "0123456789ABCDEF";
    var color = "#";
    let colors = [];
    while (colors.length < categoriesNumber) {
      for (var i = 0; i < 6; i++) {
        color += letters[Math.floor(Math.random() * 16)];
      }
      colors.push(color);
      color = "#";
    }
    this.setState({
      colors
    });
  };

  UNSAFE_componentWillReceiveProps(nextProps) {
    this.getRandomColor(nextProps.data.length);
  }

  renderCustomizedLabel = ({
    cx,
    cy,
    midAngle,
    innerRadius,
    outerRadius,
    percent,
    index
  }) => {
    const RADIAN = Math.PI / 180;
    const radius = innerRadius + (outerRadius - innerRadius) * 0.5;
    const x = cx + radius * Math.cos(-midAngle * RADIAN);
    const y = cy + radius * Math.sin(-midAngle * RADIAN);

    return (
      <text
        x={x}
        y={y}
        fill="white"
        textAnchor={x > cx ? "start" : "end"}
        dominantBaseline="central"
      >
        {`${(percent * 100).toFixed(0)}%`}
      </text>
    );
  };

  render() {
    //const COLORS = ['#0088FE', '#00C49F', '#FFBB28', '#FF8042'];
    const { classes } = this.props;
    return (
      <Paper className={classes.paper} elevation={4}>
        <h3>{this.props.title}</h3>
        <div className={classes.root}>
          <ResponsiveContainer>
            <PieChart>
              <Pie
                data={this.props.data}
                dataKey={this.props.dataKey}
                nameKey={this.props.nameKey}
                cx="50%"
                cy="40%"
                outerRadius={120}
                fill="#8884d8"
                label={this.renderCustomizedLabel}
                labelLine={false}
                unit="€"
              >
                {this.props.data.map((entry, index) => (
                  <Cell key={index} fill={this.state.colors[index]} unit="€" />
                ))}
              </Pie>
              <Tooltip />

              <Legend />
            </PieChart>
          </ResponsiveContainer>
        </div>
      </Paper>
    );
  }
}

export default withStyles(styles)(MyPieChart);
