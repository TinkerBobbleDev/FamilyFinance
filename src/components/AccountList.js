import React from 'react';
import {connect} from 'react-redux';
import AccountListItem from './AccountListItem';
import {startDeleteAccount, startAddAccount} from '../actions/accounts';
import {withStyles} from 'material-ui/styles';
import Table, {TableBody, TableCell, TableHead, TableRow} from 'material-ui/Table';

const styles = theme => ({
    root: {
        width: '100%',
        marginTop: theme.spacing.unit * 3,
        overflowX: 'auto'
    },
    table: {
        minWidth: 700
    }
});

export class AccountList extends React.Component {

    onDelete = (id) => {
        this
            .props
            .startDeleteAccount(id);
    };

    onCopy = (id) => {
        console.log(id)
        this
            .props
            .accounts
            .map((account) => {
                if (account.id === id) {
                    return this
                        .props
                        .startAddAccount(account);
                } else {
                    return console.log('no account with this ID found')
                }
            })

    };

    render() {
        const {classes} = this.props;
        return (
            <div>
                {this.props.accounts.lenght === 0
                    ? (
                        <p>no accounts</p>
                    )
                    : (
                        <Table className={classes.table}>
                            <TableHead>
                                <TableRow>
                                    <TableCell>Account</TableCell>
                                    <TableCell numeric>Balance</TableCell>
                                    <TableCell ></TableCell>

                                </TableRow>
                            </TableHead>
                            <TableBody>
                                {this
                                    .props
                                    .accounts
                                    .map((account) => {
                                        return <AccountListItem 
                                            key={account.id} 
                                            onDelete={this.onDelete}
                                            onCopy={this.onCopy}
                                            {...account}/>
                                    })
                                }
                            </TableBody>
                        </Table>
                    )
                }
            </div>

        )
    }
}

const mapStateToProps = (state, props) => {
    return {
        accounts: state.accounts,
        ...props
    };
};

const mapDispatchToProps = (dispatch, props) => ({
    startAddAccount: (account) => dispatch(startAddAccount(account)),
    startDeleteAccount: (data) => dispatch(startDeleteAccount(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(AccountList))