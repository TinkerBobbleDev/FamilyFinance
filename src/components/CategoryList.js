import React from 'react';
import {connect} from 'react-redux';
import CategoryListItem from './CategoryListItem';
import {startDeleteCategory, startAddCategory} from '../actions/categories';
import {withStyles} from 'material-ui/styles';
import Table, {TableBody, TableCell, TableHead, TableRow} from 'material-ui/Table';

const styles = theme => ({
    root: {
        width: '100%',
        marginTop: theme.spacing.unit * 3,
        overflowX: 'auto'
    },
    table: {
        minWidth: 500,
        
    }
});

export class CategoryList extends React.Component {

    onDelete = (id) => {
        this
            .props
            .startDeleteCategory(id);
    };

    onCopy = (id) => {
        console.log(id)
        this
            .props
            .categories
            .map((category) => {
                if (category.id === id) {
                    return this
                        .props
                        .startAddCategory(category);
                } else {
                    return console.log('no category with this ID found')
                }
            })

    };

    render() {
        const {classes} = this.props;
        return (
            <div>
                {this.props.categories.lenght === 0
                    ? (
                        <p>no categories</p>
                    )
                    : (
                        <Table className={classes.table}>
                            <TableHead>
                                <TableRow>
                                    <TableCell>Category</TableCell>
                                    <TableCell ></TableCell>
                                </TableRow>
                            </TableHead>
                            <TableBody>
                                { this
                                    .props
                                    .categories
                                    .map((category) => {
                                        return <CategoryListItem 
                                            key={category.id} 
                                            onDelete={this.onDelete}
                                            onCopy={this.onCopy}
                                            {...category}/>
                                    })
                                }
                            </TableBody>
                        </Table>
                    )
                }
            </div>

        )
    }
}

const mapStateToProps = (state, props) => {
    return {
        categories: state.categories,
        ...props
    };
};

const mapDispatchToProps = (dispatch, props) => ({
    startAddCategory: (category) => dispatch(startAddCategory(category)),
    startDeleteCategory: (data) => dispatch(startDeleteCategory(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(CategoryList))