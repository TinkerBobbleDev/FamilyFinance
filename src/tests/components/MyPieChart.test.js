import React from "react";
// import ReactShallowRenderer from "react-test-renderer/shallow";
import { MyPieChart } from "../../components/MyPieChart";
import toJSON from "enzyme-to-json";
import { createShallow } from "@material-ui/core/test-utils";
import { styles } from "../../components/MyPieChart";

describe("<MyPieChart />", () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow();
  });

  it("should work", () => {
    const wrapper = shallow(
      <MyPieChart
        classes={styles}
        data={[]}
        dataKey="MyPieChart"
        nameKey="MyPieChart"
      />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
