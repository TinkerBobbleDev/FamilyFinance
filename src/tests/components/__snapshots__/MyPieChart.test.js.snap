// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyPieChart /> should work 1`] = `
<WithStyles(Paper)
  elevation={4}
>
  <h3 />
  <div>
    <ResponsiveContainer
      debounce={0}
      height="100%"
      width="100%"
    >
      <PieChart
        barCategoryGap="10%"
        barGap={4}
        cx="50%"
        cy="50%"
        endAngle={360}
        innerRadius={0}
        layout="centric"
        margin={
          Object {
            "bottom": 5,
            "left": 5,
            "right": 5,
            "top": 5,
          }
        }
        outerRadius="80%"
        reverseStackOrder={false}
        stackOffset="none"
        startAngle={0}
      >
        <Pie
          animationBegin={400}
          animationDuration={1500}
          animationEasing="ease"
          blendStroke={false}
          cx="50%"
          cy="40%"
          data={Array []}
          dataKey="MyPieChart"
          endAngle={360}
          fill="#8884d8"
          hide={false}
          innerRadius={0}
          isAnimationActive={true}
          label={[Function]}
          labelLine={false}
          legendType="rect"
          minAngle={0}
          nameKey="MyPieChart"
          outerRadius={120}
          paddingAngle={0}
          startAngle={0}
          stroke="#fff"
          unit="â‚¬"
        />
        <Tooltip
          active={false}
          animationDuration={400}
          animationEasing="ease"
          contentStyle={Object {}}
          coordinate={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          cursor={true}
          cursorStyle={Object {}}
          filterNull={true}
          isAnimationActive={true}
          itemSorter={[Function]}
          itemStyle={Object {}}
          labelStyle={Object {}}
          offset={10}
          separator=" : "
          useTranslate3d={false}
          viewBox={
            Object {
              "x1": 0,
              "x2": 0,
              "y1": 0,
              "y2": 0,
            }
          }
          wrapperStyle={Object {}}
        />
        <Legend
          align="center"
          iconSize={14}
          layout="horizontal"
          verticalAlign="bottom"
        />
      </PieChart>
    </ResponsiveContainer>
  </div>
</WithStyles(Paper)>
`;
